src/components/containers/partitioned_vector/CMakeLists.txt

```diff
@@ -1,11 +1,13 @@
-# Copyright (c) 2015 Hartmut Kaiser
+# Copyright (c) 2015-2017 Hartmut Kaiser
 #
 # Distributed under the Boost Software License, Version 1.0. (See accompanying
 # file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 ###############################################################################
 set(root "${PROJECT_SOURCE_DIR}/hpx/components/containers/partitioned_vector")
 
+add_definitions(-DHPX_PARTITIONED_VECTOR_MODULE_EXPORTS)
```

Comment:
Can we have this definition set per target instead, something like:
`target_compile_definitions(partitioned_vector_component PRIVATE HPX_PARTITIONED_VECTOR_MODULE_EXPORTS)`?

Reply:
Isn't that set for this target only?

Reply:
As it is right now, yes. `add_definitions` adds it to the current `CMakeLists.txt` and anything below. In some way it is setting it globally.